From: Samuel Thibault <samuel.thibault@ens-lyon.org>
Subject: [PATCH] setitimer.c: Fix unlock in error patch

In some error path, setitimer_locked would not perform unlocking, here
is a patch.

2009-02-01  Samuel Thibault  <samuel.thibault@ens-lyon.org>

        * sysdeps/mach/hurd/setitimer.c (setitimer_locked): Use common exit path
        instead of returning without unlocking.

---
http://sourceware.org/ml/libc-alpha/2009-02/msg00003.html

 sysdeps/mach/hurd/setitimer.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/sysdeps/mach/hurd/setitimer.c b/sysdeps/mach/hurd/setitimer.c
index fec64a8..7de968d 100644
--- a/sysdeps/mach/hurd/setitimer.c
+++ b/sysdeps/mach/hurd/setitimer.c
@@ -220,7 +220,7 @@ setitimer_locked (const struct itimerval *new, struct itimerval *old,
 	  /* Start up the itimer thread running `timer_thread' (below).  */
 	  if (err = __thread_create (__mach_task_self (),
 				     &_hurd_itimer_thread))
-	    return __hurd_fail (err);
+	    goto out;
 	  _hurd_itimer_thread_stack_base = 0; /* Anywhere.  */
 	  _hurd_itimer_thread_stack_size = __vm_page_size; /* Small stack.  */
 	  if (err = __mach_setup_thread (__mach_task_self (),
-- 
tg: (0234227..) t/itimer-lock (depends on: baseline)
